happi/cli.py

```diff
@@ -132,8 +132,16 @@ def search(
             continue
 
         elif is_number(value):
-            logger.debug('Changed %s to float', value)
-            # value = str(float(value))
+            if float(value) == int(float(value)):
+                # value is an int, allow the float version (optional .0)
+                logger.debug(f'looking for int value: {value}')
+                value = f'^{int(float(value))}(\\.0+$)?$'
```

Comment:
I think this is good as you have it: 
* Requiring it to match the entire integer value - as marked by `^...$`
* With or without the trailing `.0`.